iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -1030,6 +1030,45 @@ static void copyFromAliasingBufferToResultBuffer(OpBuilder &b, Location loc,
   }
 }
 
+static LogicalResult convertPadTensorOp(
```

Comment:
There are a few missing pieces here, that have to do with the overall flow of this pass. (following what is done in `convertAnyLinalgOp` might help here and I did try to document what this pass does, so please let me know if something is not clear).

- You need to add a method to `analyseOperations` to analyse `linalg.pad_tensor`. There are two options here
  - If the source and the dest of the `linalg.pad_tensor` are the same, then there is no padding needed. in that cause you can "tie" the result of the pad tensor to the source, to indicate that the result can reuse the buffer for the source)
  - You can also just add a canonicalization for `linalg.pad_tensor` to drop it when there is actually no padding.
  In any case there should be an entry in `analyseOperations` for completeness. 
- There is a general logic in `getOrAllocateBuffer` that understands "tied" operands. That will allocate a buffer if needed automatically or return an alias if possible. It will update the `BlockAndValueMapping` with the buffer (so result is already mapped).
- Here you can just do the copy.

We can chat about this if you want a bit more. I think though landing this commit (https://github.com/MaheshRavishankar/iree/commit/a42354b4cb8dd55022775e19294be9536b91b628) would make thigns a bit easier. Then the conversion here is just a special case  of the conversion at line 1146.

Reply:
> There are a few missing pieces here, that have to do with the overall flow of this pass. (following what is done in `convertAnyLinalgOp` might help here and I did try to document what this pass does, so please let me know if something is not clear).
> 
> * You need to add a method to `analyseOperations` to analyse `linalg.pad_tensor`. There are two options here
>   
>   * If the source and the dest of the `linalg.pad_tensor` are the same, then there is no padding needed. in that cause you can "tie" the result of the pad tensor to the source, to indicate that the result can reuse the buffer for the source)
>   * You can also just add a canonicalization for `linalg.pad_tensor` to drop it when there is actually no padding.
>     In any case there should be an entry in `analyseOperations` for completeness.

That the plan when padding to the same original size it gets canonicalized away. I am worried adding such more things like that here the  `analyseOperations` will grow to include many similar canonicalizations.

> * There is a general logic in `getOrAllocateBuffer` that understands "tied" operands. That will allocate a buffer if needed automatically or return an alias if possible. It will update the `BlockAndValueMapping` with the buffer (so result is already mapped).
> * Here you can just do the copy.

Make sense, will up update the PR to use `getOrAllocateBuffer`

> We can chat about this if you want a bit more. I think though landing this commit ([MaheshRavishankar@a42354b](https://github.com/MaheshRavishankar/iree/commit/a42354b4cb8dd55022775e19294be9536b91b628)) would make thigns a bit easier. Then the conversion here is just a special case of the conversion at line 1146.